components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -984,8 +984,25 @@ void RewardsServiceImpl::FetchWalletProperties() {
 
 void RewardsServiceImpl::FetchGrant(const std::string& lang,
     const std::string& payment_id) {
-  ledger_->FetchGrant(lang, payment_id);
+#if !defined(OS_ANDROID)
+  ledger_->FetchGrant(lang, payment_id, "");
+#else
+  safetynet_check::ClientAttestationCallback attest_callback = base::BindOnce(&RewardsServiceImpl::FetchGrantAttestationResult, base::Unretained(this),
+                                                                                lang, payment_id);
+  safetynet_check_runner_.performSafetynetCheck("", std::move(attest_callback));
+#endif
+}
+
+#if defined(OS_ANDROID)
+void RewardsServiceImpl::FetchGrantAttestationResult(const std::string& lang, const std::string& payment_id,
+                                                      bool result, const std::string& result_string) {
+  if (result) {
+    ledger_->FetchGrant(lang, payment_id, result_string);
+  } else {
+    LOG(ERROR) << "FetchGrantAttestationResult error: " << result_string;
```

Comment:
the same as @bridiver commented below, we need to report the error back with OnGrantCaptcha